tests/returntypes/test_new_return_types_qnode.py

```diff
@@ -188,117 +366,1320 @@ def circuit(x):
             qml.CRX(x, wires=[0, 1])
             return qml.probs(op=op, wires=wires)
 
-        qnode = qml.QNode(circuit, dev)
-        res = qnode(0.5)
+        qnode = qml.QNode(circuit, dev, diff_method=None)
+        res = qnode(tf.Variable(0.5))
+
+        if wires is None:
+            wires = op.wires
+
+        assert res.shape == (2 ** len(wires),)
+        assert isinstance(res, tf.Tensor)
+
+    @pytest.mark.parametrize("device", devices)
+    @pytest.mark.parametrize(
+        "measurement", [qml.sample(qml.PauliZ(0)), qml.sample(wires=[0]), qml.sample(wires=[0, 1])]
+    )
+    def test_sample(self, measurement, device, shots=100):
+        """Test the sample measurement."""
+        import tensorflow as tf
+
+        if device == "default.mixed" or "default.qubit":
+            pytest.skip("Sample need to be rewritten for Tf.")
```

Comment:
What does this refer to?

Reply:
Here we have problem multiple problems with sample which does not work with asarray. I think I will open a PR to fix those problems. Also for shot vectors, TF does not support array.T in statistics_new. I will update both in the next PR.